lib/active_merchant/billing/gateways/trans_first_transaction_express.rb

```diff
@@ -537,14 +537,16 @@ def add_contact(doc, fullname, options)
           doc["v1"].title options[:title] if options[:title]
 
           if (billing_address = options[:billing_address])
-            doc["v1"].phone do
-              doc["v1"].type (options[:phone_number_type] || "4")
-              doc["v1"].nr billing_address[:phone].gsub(/\D/, '') if billing_address[:phone]
+            if billing_address[:phone]
+              doc["v1"].phone do
+                doc["v1"].type (options[:phone_number_type] || "4")
+                doc["v1"].nr billing_address[:phone].gsub(/\D/, '')
+              end
             end
             doc["v1"].addrLn1 billing_address[:address1]
```

Comment:
Note: Address and Zip Code are required to participate in the address verification service (AVS) and highly recommended for appropriate card not present qualification.

From reading their docs its conditional for just a regular credit, with the stipulation above. Not sure if if best to also pass it only when given so that non-AVS calls can be made.